File: backend/src/main/java/com/backend/backend/entity/Screening.java
package com.backend.backend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("screening")
public class Screening {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long movieId;
    private Long roomId;
    private Long cinemaId; // 冗余字段，方便查询

    private LocalDateTime startTime;
    private LocalDateTime endTime;

    private BigDecimal price;

    @TableField(value = "status")
    private ScreeningStatus status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    // 定义场次状态枚举
    public enum ScreeningStatus {
        PENDING_APPROVAL, // 待审批
        APPROVED,         // 已批准
        REJECTED,         // 已拒绝
        FINISHED,         // 已结束 (可以通过定时任务或查询时判断 endTime)
        CANCELLED         // 已取消
    }
} 

File: frontend/src/types/screening.ts
export interface Screening {
  id: number
  movieId: number
  roomId: number
  cinemaId: number
  startTime: string // ISO 8601 date-time string
  endTime: string // ISO 8601 date-time string
  price: number
  status: 'PENDING_APPROVAL' | 'APPROVED' | 'REJECTED' | 'FINISHED' | 'CANCELLED'
  createTime?: string // Optional, as not always present in list views
  updateTime?: string // Optional

  // Potential fields added by backend (if backend joins data)
  movieTitle?: string
  cinemaName?: string
  roomName?: string
}

/**
 * 场次详情视图对象 (用于选座页)
 * 来自 GET /api/screenings/{screeningId}
 */
export interface ScreeningDetailVO {
  screeningId: number
  movieId: number
  movieTitle: string
  moviePosterUrl?: string // 可选
  movieDuration?: number // 可选
  cinemaId: number
  cinemaName: string
  roomId: number
  roomName: string
  startTime: string // ISO 8601 string
  endTime: string // ISO 8601 string
  price: number
  roomRowsCount?: number // 可选
  roomColsCount?: number // 可选
  status: 'PENDING_APPROVAL' | 'APPROVED' | 'REJECTED' | 'FINISHED' | 'CANCELLED'
}
